etc/ci_scripts/dump_artifacts.py

```diff
@@ -14,77 +17,135 @@
+
+    Note, however, that these logs will be only what the lambda
+    function itself outputs during its execution. There will still be
+    important information in the Localstack logs concerning how the
+    functions are *invoked* that may be useful in debugging issues.
+
+    """
+    destination = dir / f"{lambda_name}_lambda.log"
+    logging.info(f"Dumping logs for '{lambda_name}' lambda function to '{destination}'")
     with open(destination, "wb") as out_stream:
-        popen = subprocess.Popen(
+        subprocess.run(
             [
-                "docker",
+                "aws",
+                "--endpoint-url=http://localhost:4566",
                 "logs",
-                "--timestamps",
-                container_name,
+                "tail",
+                f"/aws/lambda/{lambda_name}"
             ],
             stdout=out_stream,
+            env={
+                "PATH": os.environ["PATH"],
+                # "test" is the value assumed by Localstack
+                "AWS_ACCESS_KEY_ID": "test",
+                "AWS_SECRET_ACCESS_KEY": "test"
+            }
         )
-        popen.wait()
 
 
-ARTIFACTS_PATH = Path("/tmp/dumped_artifacts").resolve()
+def _dump_docker_log(container_name: str, dir: Path) -> None:
+    """
+    run `docker logs` and dump to $DIR/$CONTAINER_NAME.log
+    """
+    destination = dir / f"{container_name}.log"
+    logging.info(f"Dumping logs for '{container_name}' container to '{destination}'")
+    with open(destination, "wb") as out_stream:
+        subprocess.run(
+            f"docker logs --timestamps {container_name}",
+            stdout=out_stream,
+            shell=True,
+        )
 
 
-def dump_all_logs(compose_project: str) -> None:
-    containers = _name_of_all_containers(compose_project)
-    os.makedirs(ARTIFACTS_PATH, exist_ok=True)
+def dump_all_logs(compose_project: str, artifacts_dir: Path) -> None:
+    containers = _container_names_by_prefix(compose_project)
     for container in containers:
-        _dump_docker_log(container_name=container, dir=ARTIFACTS_PATH)
+        _dump_docker_log(container_name=container, dir=artifacts_dir)
+    for lambda_fn in _lambda_names():
+        _dump_lambda_log(lambda_fn, dir=artifacts_dir)
 
 
-def dump_volume(compose_project: Optional[str], volume_name: str) -> None:
+def dump_volume(compose_project: Optional[str], volume_name: str, artifacts_dir: Path) -> None:
     # Make a temporary container with the volume mounted
     # docker-compose prefixes volume names with the compose project name.
     prefix = f"{compose_project}_" if compose_project else ""
-    cmd = f"docker run -d --volume {prefix}{volume_name}:/{volume_name} alpine true"
     container_id = (
-        subprocess.run(cmd.split(" "), capture_output=True)
-        .stdout.decode("utf-8")
-        .strip()
+        subprocess.run(
+            f"docker run -d --volume {prefix}{volume_name}:/{volume_name} alpine true",
+            shell=True,
```

Comment:
i finally learned about what `shell=True` and how that interacts with spaces-in-a-cmd last night :+1: